src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -158,20 +176,44 @@ public async Task<bool> UIRestorePackagesAsync(CancellationToken token)
             {
                 _packageRestoreManager.PackageRestoreFailedEvent += PackageRestoreFailedEvent;
                 string solutionDirectory = await _solutionManager.GetSolutionDirectoryAsync(token);
-                await _packageRestoreManager.RestoreMissingPackagesInSolutionAsync(solutionDirectory,
-                    this,
-                    new LoggerAdapter(this),
-                    token);
 
-                if (_restoreException == null)
+                if (_isAssetsFileMissing)
                 {
-                    // when the control is first loaded, check for missing packages
-                    await _packageRestoreManager.RaisePackagesMissingEventForSolutionAsync(solutionDirectory, token);
+                    var componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                    _solutionRestoreWorker = new Lazy<ISolutionRestoreWorker>(
+                        () => componentModel.GetService<ISolutionRestoreWorker>());
+
+                    await _solutionRestoreWorker.Value.ScheduleRestoreAsync(
```

Comment:
ARe we restoring the full solution on purpose here? 
Would we be better off restore just that project?

We should validate that with PM.

Reply:
Yes, we decided with @chgill-MSFT that restoring all the solution was the best solution for the user. This banner will be used in rare occasions (for example a git clean or freshly cloned project, and the user has disabled the auto restore option)

Reply:
Sorry I was referring to the options in VS about restore 
![image](https://user-images.githubusercontent.com/43253759/154552003-6985f55d-3892-43dd-9a50-4ec2a3fb01b6.png)

Reply:
That disables restore altogether. 
If that's disabled, you can't restore at all.

Reply:
Yess, the scenario that I'm thinking of is: The user clones/cleans a repo and doesn't have the assets file, if he opens the pm ui we will show the banner and if the user restores it will fail and show an error message to disable that option.